src/ActionList/Item.tsx

```diff
@@ -0,0 +1,80 @@
+import type {IconProps} from '@primer/octicons-react'
+import React from 'react'
+import styled from 'styled-components'
+import {get} from '../constants'
+
+interface ItemPropsBase extends React.ComponentPropsWithoutRef<'div'> {
```

Comment:
I'm going to write some TypeScript authoring guidelines but I think we should avoid including `React.ComponentProps<>` in the props types that we export. This will make it easier for consumers to use these types with the `as` prop. 

For example, if someone wanted to make a wrapper around ActionList.Item that rendered a button, the `div` props could make things tricky:

```
import {ActionList, ActionListItemProps} from '@primer/components'

type MyItemProps = ActionListItemProps & React.ComponentPropsWithoutRef<'button'>
// Now MyItemProps has button and div props :(

function MyItem(props: MyItemProps) {
  return <ActionList.Item {...props} as="button" />
}
```

```suggestion
interface ItemPropsBase {
```

Reply:
I’m definitely open to removing `extends React.ComponentPropsWithoutRef<'div'>`. Re: the `"as"` prop, we may want to disallow it for `ActionList.Item`s, since they can contain multiple interactive elements (example below), but `button`’s can’t (i.e. nested `button`s is [not valid HTML](https://validator.w3.org/nu/#textarea)).

@vdepizzol and I discussed examples where the `ActionList` pattern (if not this PR’s implementation) can appear outside `DropdownMenu` and `ActionMenu`. One such example is the issues/PRs list (screenshot below). Items in these lists include titles, labels, and authors; clicking these initiates three different actions. If `ActionList.Item`s were rendered `as` a `button` (or an `a`), the rendered HTML would be invalid.

<img width="1217" alt="A list of Primer Components PRs" src="https://user-images.githubusercontent.com/3104489/113090935-e4965700-91b8-11eb-8c8a-51dd6bb39915.png">